clients/csharp-uwp/UWPVoiceAssistantSample/MainPage.xaml.cs

```diff
@@ -252,38 +264,102 @@ private async Task UpdateUIForSharedStateAsync()
                 var microphoneStatusInfo = await UIAudioStatus.GetMicrophoneStatusAsync();
                 this.MicrophoneInfoIcon.Glyph = microphoneStatusInfo.Glyph;
                 this.MicrophoneInfoIcon.Foreground = new SolidColorBrush(microphoneStatusInfo.Color);
-                this.MicrophoneLinkButton.Content = microphoneStatusInfo.Status;
+
+                int teachingTipCount = 0;
+
+                if (microphoneStatusInfo.Status[0] == "Microphone is available.")
+                {
+                    this.MicrophoneLinkButton.Content = "Microphone is available.";
+                    this.TeachingTipStackPanel.Children.Clear();
+                }
+                else
+                {
+                    this.MicrophoneLinkButton.Content = "Microphone is not available.";
+                    this.VAStatusIcon.Glyph = Glyphs.Warning;
+                    this.TeachingTipStackPanel.Children.Clear();
+                }
 
                 var voiceActivationStatusInfo = await UIAudioStatus.GetVoiceActivationStatusAsync();
                 this.VAStatusIcon.Glyph = voiceActivationStatusInfo.Glyph;
                 this.VAStatusIcon.Foreground = new SolidColorBrush(voiceActivationStatusInfo.Color);
-                this.VoiceActivationLinkButton.Content = voiceActivationStatusInfo.Status;
 
-                this.DismissButton.Visibility = session.IsUserAuthenticated ? Visibility.Collapsed : Visibility.Visible;
+                if (voiceActivationStatusInfo.Status[0] == "Voice activation is configured and available.")
```

Comment:
If you are going to use strings to check this, I'd prefer you made them consts in case we ever change them. ie

`private static readonly string voiceActivationEnabledMessage = "Voice activation is configured and available."`

then use voiceActivationEnabledMessage everywhere you currently use the string. Same with the mic enabled string.

Reply:
Done - did the same for "microphone is available."